src/Common/src/System/Net/SocketAddressPal.Unix.cs

```diff
@@ -115,11 +120,16 @@ public static unsafe uint GetIPv4Address(byte[] buffer)
             return ipAddress;
         }
 
-        public static unsafe void GetIPv6Address(byte[] buffer, Span<byte> address, out uint scope)
+        public static void GetIPv6Address(byte[] buffer, Span<byte> address, out uint scope)
+        {
+            GetIPv6Address(new ReadOnlySpan<byte>(buffer), address, out scope);
+        }
```

Comment:
Do we need this overload?  I'd expect the call site passing in the byte[] to just work with the span-based overload due to the implicit array->span cast.